admin/cli.py

```diff
@@ -69,6 +69,46 @@ def get_tunnel(tunnel_id) -> Optional[dict]:
     return None
 
 
+def connect_tunnel(c, tunnel_id):
+    """Connects to a live support tunnel using the tunnel_id, and returns a connection to the device"""
+    # Care should be exercised here; we're taking data from a remote source and using it to
+    # run shell commands. Validate every last bit of data.
+    t = get_tunnel(tunnel_id)
+    assert TunnelState(t['state']) == TunnelState.running, "Device has not yet connected"
+    dip = device_ip(IPv4Network(t['network'])).ip
+    assert t['support_user'].isalnum()
+    assert t['support_user'].isascii()
+    support_user = t['support_user']
+
+    # set up local
+    c.run("gcloud compute config-ssh", hide="both")
+    user_from_oslogin = c.run("gcloud compute os-login describe-profile --format=json", hide="both")
+    ts_user = json.loads(user_from_oslogin.stdout)['posixAccounts'][0]['username']
+
+    # set up connection to bastion
+    ts = Connection(
+        host = str(get_ts_instance_public_ip(tunnel_id)),
+        user = ts_user,
+        connect_kwargs={"auth_timeout": 120}
+    )
+
+    # grab the ssh private key on the tunnel server
+    ssh_privkey = ts.run(f"sudo cat {SSH_KEYFILE_PATH}", hide="both")
+    assert ssh_privkey
+
+    # set up connection to destination device
+    device = Connection(
+        host=str(dip),
+        user=support_user,
+        gateway=ts,
+        connect_kwargs = {
+            "pkey": Ed25519Key.from_private_key(io.StringIO(ssh_privkey.stdout)),
+        }
+    )
+
+    return device
```

Comment:
Moved a majority of the `fab connect` task to its own function, as that was being reused by my new tasks to access the remote device
This causes a new auth token verification to be required for every task that uses this function, which could exhaust the 2FA limit that google has before they cut you off for a few hours. It'd be nicer to find a secure way to store the session data to be reused instead but I'll need help with that if we're interested in going down that path